### PR TITLE
[messageui] Update for iOS 10 beta 1 & 3

### PR DESCRIPTION
**Pull request by VincentDondain**

### CODE REVIEW

File: src/MessageUI/MessageUI.cs [hunk 1]
```diff
@@ -12,22 +12,19 @@
 
 namespace XamCore.MessageUI {
 
-	// untyped enum -> MFMailComposeViewController.h
```
Comment (spouliot) [inline src/MessageUI/MessageUI.cs]: there's no reason to remove the comments, they make it clear they should not be `[Native]`

**Comment by xamarin-release-manager** [comment]
Build failure

File: src/MessageUI/MessageUI.cs (hunk 1 above)
Comment (spouliot) [inline src/MessageUI/MessageUI.cs]: If they were changed to `NSInteger` then it's a breaking change and we'll need to review the API using them (as we can't change the public `enum` size).

The comment will need to be updated, but not just removed.

Comment (VincentDondain) [inline src/MessageUI/MessageUI.cs]: Yes I just realized the mistake, should be `nint`.

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by VincentDondain** [comment]
List of affected APIs:

- `MFMailComposeViewControllerDelegate` https://github.com/xamarin/xamarin-macios/blob/master/src/messageui.cs#L56-L60
- `MFMessageComposeViewControllerDelegate ` https://github.com/xamarin/xamarin-macios/blob/master/src/messageui.cs#L139-L144
- `MFMessageComposeResultEventArgs ` https://github.com/xamarin/xamarin-macios/blob/master/src/MessageUI/MessageUI2.cs#L69 & https://github.com/xamarin/xamarin-macios/blob/master/src/MessageUI/MessageUI2.cs#L74
- private `Mono_MFMessageComposeViewControllerDelegate` https://github.com/xamarin/xamarin-macios/blob/master/src/MessageUI/MessageUI2.cs#L110-L114

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by spouliot** [comment]
build (timed out previously)

@VincentDondain let's open a bug with the above information so we can apply this PR, which is fine (even better) otherwise.

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by VincentDondain** [comment]
@spouliot bug filed: https://bugzilla.xamarin.com/show_bug.cgi?id=42682

**Pull request merged**